### PR TITLE
[R][Client] Fix api response, JSON for maps and let httr2 api client handle empty response bodies

### PR DESCRIPTION
**Pull request by joXemMx**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

**ApiResponse fix**: 
The current code for `response_to_text` is:
```
response_as_text = function(from_encoding = NULL, to_encoding = "UTF-8") {
      text_response <- iconv(readBin(self$response, character()), from = from_encoding, to = to_encoding)
      if (is.na(text_response)) {
        warning("The response is binary and will not be converted to text.")
      }
      return(text_response)
    }
```
But `from_encoding = NULL` is invalid and produces an error, as only strings are allowed, see `iconv` docs https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/iconv. I changed it to an empty string, which works as intended:
```
response_as_text = function(from_encoding = "", to_encoding = "UTF-8") {
      text_response <- iconv(readBin(self$response, character()), from = from_encoding, to = to_encoding)
      if (is.na(text_response)) {
        warning("The response is binary and will not be converted to text.")
      }
      return(text_response)
    }
```

**JSON for maps fix**:
Currently maps are generated like
```
"supportedILPSolvers": "{
        "CLP": "2.10.8.2",
        "GUROBI": "9.1.1",
        "CPLEX": "12.7.1"
    }"
```
by code following this pattern (actual snippet of my generated code):

```
if (!is.null(self$`supportedILPSolvers`)) {
          sprintf(
          '"supportedILPSolvers":
            "%s"
          ',
          jsonlite::toJSON(lapply(self$`supportedILPSolvers`, function(x){ x }), auto_unbox = TRUE, digits = NA)
          )
        }
```
which is wrong JSON and produces an error in the classes `toJSONString` function.
I simply changed the template to produce `%s`  instead of `"%s"` for maps, which results in perfect JSON and removes the error. Resulting JSONs look like
```
"supportedILPSolvers": {
        "CLP": "2.10.8.2",
        "GUROBI": "9.1.1",
        "CPLEX": "12.7.1"
    }
```

**httr2 api_client handling empty response bodies**:
As it is currently, an empty response body will lead to 
```
Backtrace:
    ▆
 1. ├─api_instance$CloseProjectSpace(project_id)
 2. │ └─self$CloseProjectSpaceWithHttpInfo(project_id, ...) at generated/R/projects_api.R:492:6
 3. │   └─self$api_client$CallApi(...) at generated/R/projects_api.R:539:6
 4. │     └─self$Execute(...) at generated/R/api_client.R:153:6
 5. │       └─resp %>% resp_body_raw() at generated/R/api_client.R:263:8
 6. └─httr2::resp_body_raw(.)
 7.   └─rlang::abort("Can not retrieve empty body")
```
in the httr2 `api_client.R`. I simply rewrote this to take the value of NULL instead of calling `resp %>% resp_body_raw()` when the response body is empty.

Mentioning technical staff: @Ramanth @saigiridhar21

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/r/modelGeneric.mustache [hunk 1]
```diff
@@ -330,7 +330,7 @@
           {{/isArray}}
           {{#isMap}}
           {{#isPrimitiveType}}
-            {{#isNumeric}}%d{{/isNumeric}}{{^isNumeric}}{{^isBoolean}}"{{/isBoolean}}%s{{^isBoolean}}"{{/isBoolean}}{{/isNumeric}}
+            {{#isNumeric}}%d{{/isNumeric}}{{^isNumeric}}{{^isBoolean}}{{/isBoolean}}%s{{^isBoolean}}{{/isBoolean}}{{/isNumeric}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/r/modelGeneric.mustache:333]: thanks for the PR. for this change, can we add a test model in the test spec to ensure it's tested moving forward?

Comment (joXemMx) [reply modules/openapi-generator/src/main/resources/r/modelGeneric.mustache:333]: Please clarify what you need, or link a PR that did something similar. I would be glad to contribute some testing!

Comment (wing328) [reply modules/openapi-generator/src/main/resources/r/modelGeneric.mustache:333]: please update modules/openapi-generator/src/test/resources/3_0/r/petstore.yaml with test schemas (models) to ensure the above change works as expected (and then regenerate the samples)

Comment (joXemMx) [reply modules/openapi-generator/src/main/resources/r/modelGeneric.mustache:333]: I added a minimal map schema that leads to the generation of the `toJSONString` function I mentioned above. Please see if this is what you wanted.

**Comment by wing328** [comment]
lgtm. lets give it a try

**Pull request merged**